backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/RenderingController.java

```diff
@@ -15,27 +23,51 @@
 public class RenderingController {
 
   private final TransformLegalDocMlToHtmlUseCase transformLegalDocMlToHtmlUseCase;
+  private final ApplyPassiveModificationsUseCase applyPassiveModificationsUseCase;
 
-  public RenderingController(TransformLegalDocMlToHtmlUseCase transformLegalDocMlToHtmlUseCase) {
+  public RenderingController(
+      TransformLegalDocMlToHtmlUseCase transformLegalDocMlToHtmlUseCase,
+      ApplyPassiveModificationsUseCase applyPassiveModificationsUseCase) {
     this.transformLegalDocMlToHtmlUseCase = transformLegalDocMlToHtmlUseCase;
+    this.applyPassiveModificationsUseCase = applyPassiveModificationsUseCase;
   }
 
   /**
-   * Retrieves the HTML rendering of a law document.
+   * Retrieves the HTML rendering of a norm.
    *
-   * @param xml The XML content of the law document to be rendered.
+   * @param previewRequestSchema The Request schema for rendering a norm. It includes the norm xml
+   *     and any additional norms that should be used instead of the saved once for rendering the
+   *     norm.
    * @param showMetadata A boolean indicating whether metadata should be included in the rendering.
+   * @param atIsoDate ISO date string indicating which modifications should be applied before the
+   *     HTML gets rendered and returned. If no date is provided the current date is used.
    * @return A {@link ResponseEntity} containing the HTML rendering of the law document.
    */
   @PostMapping(
-      consumes = {APPLICATION_XML_VALUE},
+      consumes = {APPLICATION_JSON_VALUE},
       produces = {TEXT_HTML_VALUE})
   public ResponseEntity<String> getHtmlPreview(
-      @RequestBody final String xml, @RequestParam boolean showMetadata) {
+      @RequestBody final PreviewRequestSchema previewRequestSchema,
+      @RequestParam(defaultValue = "false") boolean showMetadata,
+      @RequestParam Optional<Instant> atIsoDate) {
+    final var norm =
+        Norm.builder().document(XmlMapper.toDocument(previewRequestSchema.getNorm())).build();
+    final Map<String, Norm> customNorms =
```

Comment:
Depending on what you decide about if we need the pairs ELI + XML this would be much easier.